site/en/docs/extensions/reference/userscripts/index.md

```diff
@@ -0,0 +1,102 @@
+    width="400", height="183",  class="screenshot" %}
+
+      <figcaption>
+      Extensions page (chrome://extensions)
+      </figcaption>
+    </figure>
+
+You can determine if developer mode is enabled by checking whether `chrome.userScripts` throws an error.
+
+```javascript
+function isUserScriptsAvailable() {
+  try {
+    // Property access which throws if developer mode is not enabled.
+    chrome.userScripts;
+    return true;
+  } catch {
+    // Not available.
+    return false;
+  }
+}
+```
+
+### Work in isolated worlds
+
+Both user and content scripts can run in an isolated world or in the main world. An isolated world is an execution environment that isn't accessible to a host page or other extensions. This lets a user script change its JavaScript environment without affecting the host page or other extensions' user and content scripts. Conversely, user scripts (and content scripts) are not visible to the host page or the user and content scripts of other extensions. Scripts running in the main world are accessible to host pages and other extensions and are visible to host pages and to other extensions. To select the world, pass `"USER_SCRIPT"` or `"MAIN"` when calling `userScripts.register()`.
+
+To configure a [content security policy](https://developer.mozilla.org/docs/Web/HTTP/CSP) for whichever world you specify, call `userScripts.configureWorld()` as shown below.
+
+```javascript
+browser.userScripts.configureWorld({
+  csp: "script-src 'self'"
+});
+```
+
+### Messaging
+
+Like content scripts and offscreen documents, user scripts can communicate with other parts of an extension using [messaging](/docs/extensions/mv3/messaging/). Unlike other types of messages, user script messages use their own methods: `runtime.onUserScriptMessage` and `chrome.runtime.onUserScriptConnect` events.
+
+Before sending a message, you must call `configureWorld()` with the `messaging` argument set to `true`. Note that both the `csp` and `messaging` arguments can be passed at the same time.
+
+```javascript
+browser.userScripts.configureWorld({
+  messaging: true,
+});
+```
+
+### Extension updates
+
+User scripts are cleared when an extension updates. You can add them back running code via the [`runtime.onInstalled`](/docs/extensions/reference/runtime/#event-onInstalled) event handler in the extension service worker. Respond only to the [`"update"` reason](/docs/extensions/reference/runtime/#type-OnInstalledReason:~:text=as%20an%20installation.-,%22update%22,-Specifies%20the%20event) passed to the event callback.
+
+## Examples
+
+### Register a script
+
+The following example shows a basic call to `register()`. The first argument is an array of objects defining the scripts to be registered. There are more options than are shown here.
+
+```javascript
+chrome.userScripts.register([{
+  id: 'test',
+  matches: ['*://*/*'],
+  js: [{code: 'alert("Hi!")'}]
+}]);
+```
+
+### Configure the world
```

Comment:
It looks like we're still missing some sample code here?

Reply:
It's a heading I forgot to remove. I realized after I started writing that configure needed to go with the explanation of worlds.